src/details/ArborX_DetailsUtils.hpp

```diff
@@ -83,10 +84,12 @@ void exclusivePrefixSum(Kokkos::View<ST, SP...> const &src,
   using DeviceType = typename Kokkos::ViewTraits<DT, DP...>::device_type;
 
   auto const n = src.extent(0);
+  typename Kokkos::ViewTraits<DT, DP...>::value_type result = 0;
   ARBORX_ASSERT(n == dst.extent(0));
   Kokkos::parallel_scan(
       "exclusive_scan", Kokkos::RangePolicy<ExecutionSpace>(0, n),
-      Details::ExclusiveScanFunctor<ValueType, DeviceType>(src, dst));
+      Details::ExclusiveScanFunctor<ValueType, DeviceType>(src, dst), result);
```

Comment:
You don't need the "result" on the host for free, there is a device to host copy associated with it.

Did you consider having a function with a different name that does not lie about the hidden cost?

Reply:
Ah! I thought it was magic :) Looking at the code, literally the Cuda's `ParallelScan` and `ParallelScanWithTotal` differ only in
```c++
const int size = ValueTraits::value_size( m_functor );
DeepCopy<HostSpace,CudaSpace>( &m_returnvalue, m_scratch_space + (grid_x - 1)*size/sizeof(int), size );
```
(may I also point out that it is almost 200+ lines of duplicated code there)

Reply:
> Did you consider having a function with a different name that does not lie about the hidden cost?

Here you go.